src/Adapter/Profile/Employee/CommandHandler/ResetPasswordHandler.php

```diff
@@ -0,0 +1,160 @@
+use Shop;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Handles the command which resets employee's password.
+ *
+ * @internal
+ */
+final class ResetPasswordHandler implements ResetPasswordHandlerInterface
+{
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var Hashing
+     */
+    private $hashing;
+
+    /**
+     * @var LegacyContext
+     */
+    private $legacyContext;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     * @param Hashing $hashing
+     * @param LegacyContext $legacyContext
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        ConfigurationInterface $configuration,
+        Hashing $hashing,
+        LegacyContext $legacyContext,
+        TranslatorInterface $translator
+    ) {
+        $this->configuration = $configuration;
+        $this->hashing = $hashing;
+        $this->legacyContext = $legacyContext;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(ResetPasswordCommand $command)
+    {
+        $employee = new Employee();
+
+        // Legacy condition. Employee object is loaded in $employee->getByEmail() call, so it must go first.
+        $employeeFound = $employee->getByEmail($command->getEmail()->getValue())
+            && $employee
+            && $employee->id == $command->getEmployeeId()->getValue()
+        ;
+
+        if (!$employeeFound) {
+            throw new EmployeeNotFoundException(null, 'Employee could not be found by given email.');
+        }
+
+        $timeExpression = sprintf(
+            '%s + %s minutes',
+            (string) $employee->last_passwd_gen,
+            (string) $this->configuration->get('PS_PASSWD_TIME_BACK')
+        );
+        $canResetPassword = (strtotime($timeExpression) - time()) <= 0;
+
+        if (!$canResetPassword) {
+            throw new PasswordResetTooFrequentException('You cannot reset your password so frequently.');
+        }
+
+        if ($employee->getValidResetPasswordToken() !== $command->getResetToken()) {
+            throw new ResetPasswordTokenExpiredException('Reset password token has expired.');
+        }
+
+        $employee->passwd = $this->hashing->hash($command->getNewPlainPassword());
+        $employee->last_passwd_gen = date('Y-m-d H:i:s', time());
+
+        $params = array(
+            '{email}' => $employee->email,
+            '{lastname}' => $employee->lastname,
+            '{firstname}' => $employee->firstname,
+        );
+
+        $employeeLanguage = new Language((int) $this->legacyContext->getContext()->employee->id_lang);
+
+        $mailWasSent = Mail::send(
+            $employee->id_lang,
+            'password',
+            $this->translator->trans(
+                'Your new password',
```

Comment:
Note other things to check: which social media to keep and the number of shops.

Reply:
this message definitely needs to be modified, as well as the email template content. As I can see the related github issue is still open so I'll leave this as-is until it's resolved

Reply:
I'm a bit lost here, it's been a while since I last studied this page, and the virtual machine does not work for now so I can't test it to get more info. Could we help with the context of those two strings? Where do they appear exactly?
🙄

Reply:
@LouiseBonnard this "your new password" string appears in the title of reset password email:
https://prnt.sc/phbfiu

Reply:
Alright, thanks @rokaszygmantas. It refers to the email called **password** in the back office. I'd write _Your password has been updated_.